src/custom/hooks/useWrapCallback.ts

```diff
@@ -37,222 +28,191 @@ export enum WrapType {
   const isZero = balance && !inputAmount
 
-  const inputError = isZero ? t`Enter an amount` : !sufficientBalance ? t`Insufficient ${symbol} balance` : undefined
-
-  // Create wrap/unwrap callback if sufficient balance
-  let wrapUnwrapCallback:
-    | ((params?: Pick<OptionalApproveCallbackParams, 'useModals'>) => Promise<TransactionResponse>)
-    | undefined
-  if (sufficientBalance && inputAmount) {
-    let wrapUnwrap: () => Promise<TransactionResponse>
-    let summary: string
-    let confirmationMessage: string
-    let operationType: OperationType
-
-    if (!chainId) {
-      throw new Error('ChainId is unknown')
-    }
-
-    const amountHex = `0x${inputAmount.quotient.toString(RADIX_HEX)}`
-    if (isWrap) {
-      operationType = OperationType.WRAP_ETHER
-      wrapUnwrap = async () => {
-        const estimatedGas = await wethContract.estimateGas.deposit({ value: amountHex }).catch(_handleGasEstimateError)
-        const gasLimit = calculateGasMargin(estimatedGas)
-
-        return wethContract.deposit({ value: amountHex, gasLimit })
-      }
-      const baseSummary = t`${formatSmart(inputAmount, AMOUNT_PRECISION)} ${native} to ${wrapped}`
-      summary = t`Wrap ${baseSummary}`
-      confirmationMessage = t`Wrapping ${baseSummary}`
-    } else {
-      operationType = OperationType.UNWRAP_WETH
-      wrapUnwrap = async () => {
-        const estimatedGas = await wethContract.estimateGas.withdraw(amountHex).catch(_handleGasEstimateError)
-        const gasLimit = calculateGasMargin(estimatedGas)
-        return wethContract.withdraw(amountHex, { gasLimit })
-      }
-      const baseSummary = t`${formatSmart(inputAmount, AMOUNT_PRECISION)} ${wrapped} to ${native}`
-      summary = t`Unwrap ${baseSummary}`
-      confirmationMessage = t`Unwrapping ${baseSummary}`
-    }
-
-    const operationMessage = getOperationMessage(operationType, chainId)
-    wrapUnwrapCallback = async (params = { useModals: true }) => {
-      const { useModals } = params
-      try {
-        useModals && openTransactionConfirmationModal?.(confirmationMessage, operationType)
-        wrapAnalytics('Send', operationMessage)
-
-        const txReceipt = await wrapUnwrap()
-        wrapAnalytics('Sign', operationMessage)
-
-        addTransaction({
-          hash: txReceipt.hash,
-          summary,
-        })
-        useModals && closeModals?.()
-
-        return txReceipt
-      } catch (error) {
-        useModals && closeModals?.()
-
-        const isRejected = isRejectRequestProviderError(error)
-        const action = isRejected ? 'Reject' : 'Error'
-        wrapAnalytics(action, operationMessage)
-
-        const errorMessage = (isRejected ? 'Reject' : 'Error') + ' Signing transaction'
-        console.error(errorMessage, error)
-
-        throw typeof error === 'string' ? new Error(error) : error
-      }
-    }
+  return isZero ? t`Enter an amount` : !sufficientBalance ? t`Insufficient ${symbol} balance` : undefined
```

Comment:
while I'm fine with ternaries like this, it is sometimes more legible when in extended form e.g `if/else`

Reply:
Fixed